site/src/xServices/templateVersionEditor/templateVersionEditorXService.ts

```diff
@@ -170,9 +179,41 @@ export const templateVersionEditorMachine = createMachine(
         invoke: {
           id: "fetchVersion",
           src: "fetchVersion",
-          onDone: {
-            actions: ["assignBuild"],
-            target: "fetchResources",
+          onDone: [
+            {
+              actions: ["assignBuild"],
+              target: "promptVariables",
+              cond: "jobFailedWithMissingVariables",
+            },
+            {
+              actions: ["assignBuild"],
+              target: "fetchResources",
+            },
+          ],
+        },
+      },
+      promptVariables: {
+        initial: "loadingMissingVariables",
+        states: {
+          loadingMissingVariables: {
+            invoke: {
+              src: "loadMissingVariables",
+              onDone: {
+                actions: "assignMissingVariables",
+                target: "idle",
+              },
+            },
+          },
+          idle: {
+            on: {
+              SET_MISSING_VARIABLE_VALUES: {
+                actions: "assignMissingVariableValues",
+                target: "#templateVersionEditor.creatingBuild",
```

Comment:
Let's assume that template author adds a new variable `create_ec2_resources` (bool):
1. Click on `Build template`.
2. Pass the value for `foobar1`.
3. Click on `Submit`.

Now, the template author would like to check the template against a different value, and they can't because the value is stored. I think that we should cover this path.

Reply:
So, should we re-display the field? I think we cover one path and make the other worse. If I'm doing incremental changes and testing them, I would rather not having to add the value every single time 🤔

Reply:
If I'm using the editor to develop the template, I'd like to pass custom values. What should the user do if they make a typo? Is there any way to alter to value?

Frankly speaking, I'm wondering if it should be a popup or something else. Maybe a small embedded form with variables present all the time would be more convenient.

Reply:
The thing with the form is, we don't have those variables until we build the changes 🤔

Reply:
Maybe we could just have a simple version to "unblock" users in case this happens and after that, improve it. I'm starting to get very concerned about how big the scope for the template editor is getting to be honest.

Reply:
@bpmct What is your take on this one? Is it something that will annoy users?

Reply:
If we always want to show this, I agree to make it "static" is the best option but, my thing is where to place this because we already have a lot of elements being displayed and how to display it when an error happens so the user can notice it easily and know what to do next - since it is static we want to highlight it somehow when they need to input something there.